ibm/data_source_ibm_resource_group_test.go

```diff
@@ -0,0 +1,44 @@
+package ibm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccIBMResourceGroupDataSource_Basic(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMResourceGroupDataSourceConfigDefault(),
+			},
+			{
+				Config: testAccCheckIBMResourceGroupDataSourceConfigWithName(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.ibm_resource_group.testacc_ds_resource_group_name", "name", "default"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMResourceGroupDataSourceConfigDefault() string {
```

Comment:
Both are optional fields can you add a test case if none of them are specified

Reply:
Added testcase to handle is_default false and validating the error.